searchlib/src/vespa/searchlib/docstore/logdatastore.cpp

```diff
@@ -313,16 +313,23 @@ LogDataStore::compact(uint64_t syncToken)
     }
 }
 
+bool
+LogDataStore::isTotalDiskBloatExceeded() const {
+    const size_t diskFootPrint = getDiskFootprint();
+    const size_t maxConfiguredDiskBloat = diskFootPrint * _config.getMaxDiskBloatFactor();
+    return getDiskBloat() > maxConfiguredDiskBloat;
+}
+
 size_t
 LogDataStore::getMaxCompactGain() const
 {
+    size_t bloat = 0;
+    if ( isTotalDiskBloatExceeded() ) {
+        bloat = getDiskBloat();
```

Comment:
Consider a strategy to avoid having to call getDiskBloat() again here, e.g. return std::pair<bool, size_t> or just a size_t from a renamed isTotalDiskBloatExceeded().

Reply:
Done, in addition to avoid deadlock.